core/src/datayoga_core/write_utils.py

```diff
@@ -51,6 +52,14 @@ def map_record(record: Dict[str, Any],
 
         # columns with spaces will be later used with underscores in the bind variables
         target = (next(iter(item.keys())) if isinstance(item, dict) else item).replace(" ", "_")
-        mapped_record[target] = None if source not in record else record[source]
+
+        source_path = utils.split_field(source)
```

Comment:
when mapping/keys contain . in it, like address.id, it parses it as a nested field and takes the value from there instead of looking for "address.id" property, which will obviously fail.